src/Dispatcher.php

```diff
@@ -101,19 +102,33 @@ public function schedule( string $action, string $entity, int|string $id, array
 	}
 
 	/**
-	 * Action Scheduler callback. Sends the POST request non-blocking.
+	 * Action Scheduler callback. Sends the POST request and throws WP_Exception on failure.
+	 *
+	 * This method throws WP_Exception upon webhook delivery failure to enable Action Scheduler
+	 * to properly catch and handle the exception. This provides better visibility into failure
+	 * reasons within Action Scheduler's logging and retry mechanisms, improving transparency
+	 * and debugging capabilities for webhook processing errors.
 	 *
 	 * @param string              $url The webhook URL.
 	 * @param string              $action The action type.
 	 * @param string              $entity The entity type.
 	 * @param int|string          $id The entity ID.
 	 * @param array<string,mixed> $payload The payload data.
+	 *
+	 * @throws WP_Exception When webhook delivery fails or URL is blocked.
 	 */
 	public function process_scheduled_webhook( string $url, string $action, string $entity, $id, array $payload ): void {
 
 		// Check if this URL is blocked due to too many failures
 		if ( $this->is_url_blocked( $url ) ) {
-			return;
+			throw new WP_Exception(
+				'webhook_url_blocked',
+				sprintf(
+					/* translators: %s: webhook URL */
+					__( 'Webhook URL is blocked due to consecutive failures: %s', 'wp-webhook-framework' ),
+					$url
+				)
+			);
```

Comment:
This is still not only one message. WP_Exception extends the normal exception class and therefore accepts one parameter for the message. No error code. 

https://developer.wordpress.org/reference/classes/wp_exception/

public [__construct](https://www.php.net/manual/en/exception.construct.php)([string](https://www.php.net/manual/en/language.types.string.php) $message = "", [int](https://www.php.net/manual/en/language.types.integer.php) $code = 0, [?](https://www.php.net/manual/en/language.types.null.php)[Throwable](https://www.php.net/manual/en/class.throwable.php) $previous = [null](https://www.php.net/manual/en/reserved.constants.php#constant.null))